server/server.go

```diff
@@ -684,16 +685,15 @@ func (s *Server) SetSystemAccount(accName string) error {
 		return s.setSystemAccount(v.(*Account))
 	}
 
-	s.mu.Lock()
 	// If we are here we do not have local knowledge of this account.
 	// Do this one by hand to return more useful error.
 	ac, jwt, err := s.fetchAccountClaims(accName)
 	if err != nil {
-		s.mu.Unlock()
 		return err
 	}
 	acc := s.buildInternalAccount(ac)
 	acc.claimJWT = jwt
+	s.mu.Lock()
 	s.registerAccount(acc)
```

Comment:
I think RegisterAccount does locking so you could use that.

Reply:
No, RegisterAccount(string) accepts a string for the account name and create an account then call registerAccount(*Account).

Reply:
No if you want I could create a registerAccountLocked() (or whatever name you chose) to do this lock/registerAccount/unlock..

Reply:
I like that idea..